lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
If you think xl and xl_r should be two different compilers, then we'll need to rebase this PR on the latest develop branch and remove the xl_r names from xl.py. Shouldn't be too difficult.